src/go/rpk/pkg/cli/cmd/cloud/auth/auth.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// Package auth contain specific configuration and handlers for the
+// authentication with the Redpanda cloud.
+package auth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/auth0"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/cmd/cloud/cloudcfg"
+	rpkos "github.com/redpanda-data/redpanda/src/go/rpk/pkg/os"
+	"github.com/spf13/afero"
+)
+
+// The auth0 endpoint information to get dev tokens from.
+var prodAuth0Endpoint = auth0.Endpoint{
+	URL:      "https://prod-cloudv2.us.auth0.com",
+	Audience: "cloudv2-production.redpanda.cloud",
+}
+
+type BadClientTokenError struct {
```

Comment:
```
// BadClientTokenError is returned when the client ID is invalid or some other error occurs.
// This can be used as a hint that the client ID needs to be cleared as well.
```